core/textinput/src/textinput/Editor.cpp

```diff
@@ -262,10 +283,23 @@ namespace textinput {
         } else {
           ++fCurHistEntry;
         }
+        SetReverseHistSearchPrompt(R.fDisplay);
         if (UpdateHistSearch(R)) return kPRSuccess;
         fCurHistEntry = prevHistEntry;
         return kPRError;
-      } else {
+      } else if (M == kCmdForwardSearch) {
+         // Search again. Move to newer hist entry:
+         size_t prevHistEntry = fCurHistEntry;
+         if (fCurHistEntry == 0) {
```

Comment:
Given that `fCurHistEntry` might also be `-1`, I think this should be
```suggestion
         if (fCurHistEntry <= (size_t)0) {
```